gitgud/levels/rampup/_reversing/instructions.txt

```diff
@@ -0,0 +1,19 @@
+Sometimes you may find that you made changes that you aren't happy with. Wouldn't it be great if there was a simple way to undo all your changes and return to the last commit? Here are two ways!
+>>>
+The commands "git reset" and "git revert" both allow you to rollback changes to an older commit. While "git reset" is used for the local branch, "git revert" is used for public branches.
+>>>
+Let's start with "git reset". There are three modes: --soft, --mixed(default), and --hard.
+>>>
+The "git reset --soft <commit_ref>" is not used often. The branch pointer will move to the specified <commit_ref> (can use hash, HEAD^, HEAD~<num>, etc!). All the changes you've made since that commit will be left as staged ("Changes to be committed" in git status).
+>>>
+The "git reset --mixed <commit_ref>" is the default mode of git reset. In other words "git reset <commit_ref>" implies the mixed mode. The functionality is similar to soft mode—the branch pointer will be moved to <commit_ref> and changes will be preserved. The only difference is that your changes will be unstaged (will need to use "git add").
+>>>
+The third option is the most extreme (but may be the most useful). When you want to rollback the clock completely to a certain commit: use "git reset --hard <commit_ref>". This command will annihilate any changes you've made since the specified commit. Essentially, using this command will reset your repository to exactly what it was at the time of the specified commit. Be careful with this option!
```

Comment:
So I think we should start by reiterating that git reset throws away the commit, and then we can introduce mixed and soft together for the situation of preserving work and then telling them to use hard when they want to throw away all their changes

Examples would probably be useful, so here's the kind of thing I think we should highlight:

A typical use case for git reset --hard is when you've made changes to the wrong branch. After you move the changes, you can bring the original branch to it's original spot. You might also be working on a branch and then you make some changes that you don't like. If you don't want to throw away those changes, you can `git branch oldwork` `git reset HEAD~2`. Things like that
The situation for mixed and soft are similar, basically, you've made changes that you want to keep, but you're not sure which part of them you want to keep. Mixed is used when you mostly don't want to keep them, soft is used when you mostly do want to keep them.

There's probably a more eloquent way of saying something like that that you can weave into the instructions